src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -222,4 +250,85 @@ Device::~Device() {
     this->device = VK_NULL_HANDLE;
 }
 
+void Device::set_object_name(const std::uint64_t object, const VkDebugReportObjectTypeEXT type, const char *name) {
+#ifndef NDEBUG
+    assert(device);
+    assert(name);
+    assert(object);
+    assert(vkDebugMarkerSetObjectName);
+
+    VkDebugMarkerObjectNameInfoEXT name_info = {};
+    name_info.sType = VK_STRUCTURE_TYPE_DEBUG_MARKER_OBJECT_NAME_INFO_EXT;
+    name_info.objectType = type;
+    name_info.object = object;
+    name_info.pObjectName = name;
+
+    vkDebugMarkerSetObjectName(device, &name_info);
+#endif
+}
+
+void Device::set_object_tag(const std::uint64_t object, const VkDebugReportObjectTypeEXT type, const std::uint64_t name,
+                            const std::size_t tag_size, const void *tag) {
+#ifndef NDEBUG
+    assert(device);
+    assert(name);
+    assert(tag_size > 0);
+    assert(tag);
+    assert(vkDebugMarkerSetObjectTag);
+
+    VkDebugMarkerObjectTagInfoEXT tagInfo = {};
+    tagInfo.sType = VK_STRUCTURE_TYPE_DEBUG_MARKER_OBJECT_TAG_INFO_EXT;
+    tagInfo.objectType = type;
+    tagInfo.object = object;
+    tagInfo.tagName = name;
+    tagInfo.tagSize = tag_size;
+    tagInfo.pTag = tag;
+
+    vkDebugMarkerSetObjectTag(device, &tagInfo);
+#endif
+}
+
+void Device::bind_region(const VkCommandBuffer command_buffer, const std::string &name, const float color[4]) {
+#ifndef NDEBUG
+    assert(command_buffer);
+    assert(!name.empty());
+    assert(vkCmdDebugMarkerBegin);
+
+    VkDebugMarkerMarkerInfoEXT debug_marker = {};
+    debug_marker.sType = VK_STRUCTURE_TYPE_DEBUG_MARKER_MARKER_INFO_EXT;
+    debug_marker.color[0] = color[0];
+    debug_marker.color[1] = color[1];
+    debug_marker.color[2] = color[2];
+    debug_marker.color[3] = color[3];
+    debug_marker.pMarkerName = name.c_str();
+
+    vkCmdDebugMarkerBegin(command_buffer, &debug_marker);
+#endif
+}
+
+void Device::insert(const VkCommandBuffer command_buffer, const std::string &name, const float color[4]) {
+#ifndef NDEBUG
+    assert(command_buffer);
+    assert(!name.empty());
+    assert(vkCmdDebugMarkerInsert);
+
+    VkDebugMarkerMarkerInfoEXT debug_marker = {};
+    debug_marker.sType = VK_STRUCTURE_TYPE_DEBUG_MARKER_MARKER_INFO_EXT;
+    debug_marker.color[0] = color[0];
+    debug_marker.color[1] = color[1];
+    debug_marker.color[2] = color[2];
+    debug_marker.color[3] = color[3];
+    debug_marker.pMarkerName = name.c_str();
+
+    vkCmdDebugMarkerInsert(command_buffer, &debug_marker);
+#endif
+}
+
+void Device::end_region(const VkCommandBuffer command_buffer) {
```

Comment:
All of these function names are a bit vague, i think they would be better as `bind_debug_region`, `insert_debug_marker` and `end_debug_region`

Reply:
Right.

Reply:
Done.